addons/src/ao_tasks.c

```diff
@@ -556,15 +565,22 @@ static void create_task(AoTasks *t, GeanyDocument *doc, gint line, const gchar *
 }
 
 
-static void update_tasks_for_doc(AoTasks *t, GeanyDocument *doc)
+static gboolean update_tasks_for_doc_idle_cb(gpointer data)
 {
+	AoTasksUpdateTasksForDocArguments *arguments = data;
+	AoTasks *t = arguments->t;
```

Comment:
Will the lifetime of AoTasks object guarantee its still valid (see below)

Reply:
I think so. The `AoTasks` object exists as long as the Addons plugin is loaded, independently of opened documents and independent of whether the Tasks addon is enabled or not.

While in theory possible I think it's highly unlikely that a user will manage to unload the Addons plugin or even only to quit Geany so that the `AoTasks` object will be destroyed while an idle callback is still waiting to be executed.